README.md

```diff
@@ -210,9 +213,125 @@ func AsynchronousFriends(forUser: jimbob) -> Deferred<[Friend]> {
 }
 ```
 
+## Usage - Producer
```

Comment:
Recommend an h2 "Usage" with h3 "Producer" and "Consumer" below, rather than prefixing two h2s with Usage, to clarify the structure.

Even better: steal the "Tasks" concept from Apple's docs and be a bit more descriptive than just "Producer" (of what?), maybe something like:

```
## Tasks
### Vending a Future
### Taking Action when a Future Is Filled
### Acting when First of Several Futures Gets Filled
### Acting when All of Several Futures Have Been Filled
```

and so on.